### PR TITLE
Added configuration file for circleci, which is using a simple workfl…

### PR DESCRIPTION
**Pull request by danielksan81**
…ow to checkout and fetch dependencies, run unit tests and sample workflow with simulated backend.

Updated makefile and ci.go to make use of docker image caching provided by circleci.

Signed-off-by: Daniel Kunz <daniel.kunz@de.bosch.com>

### CODE REVIEW

**Comment by danielksan81** [comment]
I noticed that I can also remove the -checkout in the additional steps

**Comment by danielksan81** [comment]
relates to #5

**Comment by choeppler** [comment]
To be consistent with the current structure, the new commands in [build/ci.go](6/files#diff-75e934896def206c5aeb231d2f5c9cb9) should be documented in [build/doc.go](../../../blob/8a836812/build/doc.go).

**Comment by choeppler** [comment]
> 
> 
> To be consistent with the current structure, the new commands in [build/ci.go](6/files#diff-75e934896def206c5aeb231d2f5c9cb9) should be documented in [build/doc.go](../../../blob/8a836812/build/doc.go).

Additionally, this would allow to refactor the Makefile to dynamically create all targets that are available in the build module. See #8 for a possible implementation of that approach.

**Comment by danielksan81** [comment]
> > To be consistent with the current structure, the new commands in [build/ci.go](6/files#diff-75e934896def206c5aeb231d2f5c9cb9) should be documented in [build/doc.go](../../../blob/8a836812/build/doc.go).
> 
> Additionally, this would allow to refactor the Makefile to dynamically create all targets that are available in the build module. See #8 for a possible implementation of that approach.

My proposal is that we are merging this config now as a running basis to develop.
Afterwards we do the proposed refactoring.
Adding a code coverage report is a follow up in #11 .

File: build/ci.go [hunk 1]
```diff
@@ -60,19 +60,32 @@ func main() {
 	}
 
 	switch os.Args[1] {
```
Comment (choeppler) [inline build/ci.go]: Please add one-line documentation of the four new commands to build/doc.go

**Comment by danielksan81** [comment]
> LGTM. Thanks for setting up the CI infrastructure!
> 
> A couple of minor issues I'd like to have addressed before merging:
> 
> * one nit-pick: please update the commit message to reflect the guidelines linked to in [CONTRIBUTING.md](/direct-state-transfer/dst-go/blob/develop/CONTRIBUTING.md). Specifically, a subject line is missing ATM.
> * as this is your first contribution, please check whether the [NOTICE](/direct-state-transfer/dst-go/blob/develop/NOTICE) file should be updated as described [here](/direct-state-transfer/dst-go/blob/develop/CONTRIBUTING.md#maintain-copyright-holder--contributor-list).

I aligned the mentioned issues in the last commit.
My proposal is to trya stash and use last commit message.

File: NOTICE [hunk 2]
```diff
@@ -21,5 +21,5 @@
 # Please keep the list sorted.
 
 Robert Bosch LLC
-
+Daniel Kunz <daniel.kunz@de.bosch.com>
```
Comment (choeppler) [inline NOTICE]: As your employer is copyright holder for your contribution this should be
```
Robert Bosch GmbH
    Daniel Kunz <daniel.kunz@de.bosch.com>
Robert Bosch LLC
```

**Comment by choeppler** [comment]
Closes #5

**Pull request merged**